pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/ConfusingTernaryRule.java

```diff
@@ -56,15 +57,25 @@ public class ConfusingTernaryRule extends AbstractJavaRulechainRule {
     private static final PropertyDescriptor<Boolean> IGNORE_ELSE_IF = booleanProperty("ignoreElseIf")
             .desc("Ignore conditions with an else-if case").defaultValue(false).build();
 
+    private static final PropertyDescriptor<NullCheckBranch> NULL_CHECK_BRANCH = enumProperty("nullCheckBranch", NullCheckBranch.class)
```

Comment:
This should ideally be documented in the rule description too, maybe in the rule examples...

I would do this when merging